maven-plugins/enforcer-maven-plugin/src/main/java/io/helidon/build/maven/enforcer/inclusivenaming/InclusiveNamingRule.java

```diff
@@ -0,0 +1,246 @@
+        }
+        return failures;
+    }
+
+    private void processFile(FileRequest fr, List<RuleFailure> errors) {
+        try (BufferedReader bufferedReader = Files.newBufferedReader(fr.path(), StandardCharsets.UTF_8)) {
+            String line;
+            int lineNum = 0;
+            List<Term> found = new LinkedList<>();
+            while ((line = bufferedReader.readLine()) != null) {
+                // we want to start from 1
+                lineNum++;
+                for (Term term : terms) {
+                    if (line.toLowerCase().contains(term.term)) {
+                        found.add(term);
+                    }
+                }
+                if (!found.isEmpty()) {
+                    errors.add(RuleFailure.create(fr, lineNum, "inclusive naming found: " + String.join(", ", found.toString())));
+                }
+                found.clear();
+            }
+
+        } catch (IOException e) {
+            throw new EnforcerException("Failed to process file for inclusive naming: " + fr.relativePath());
+        }
+    }
+
+    private boolean include(FileRequest fr) {
+        for (FileMatcher exclude : excludes) {
+            if (exclude.matches(fr)) {
+                return false;
+            }
+        }
+
+        for (FileMatcher include : includes) {
+            if (include.matches(fr)) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    /**
+     * Fluent API builder for {@link io.helidon.build.maven.enforcer.inclusivenaming.InclusiveNamingRule}.
+     */
+    public static class Builder {
+        private InlusiveNamingConfig inclusiveNamingConfig;
+
+        private Builder() {
+        }
+
+        /**
+         * Update builder from maven configuration.
+         *
+         * @param inclusiveNamingConfig maven config
+         * @return updated builder
+         */
+        public Builder config(InlusiveNamingConfig inclusiveNamingConfig) {
+            this.inclusiveNamingConfig = inclusiveNamingConfig;
+            return this;
+        }
+
+        /**
+         * Build a new instance of {@link io.helidon.build.maven.enforcer.inclusivenaming.InclusiveNamingRule}.
+         *
+         * @return new rule
+         */
+        public InclusiveNamingRule build() {
+            return new InclusiveNamingRule(this);
+        }
+
+        private List<Term> readTerms(String file){
+            List<Term> terms = new ArrayList<>();
+            try (InputStream is = InclusiveNamingRule.class.getResourceAsStream(file)) {
+                XmlInclusiveNaming xml = xmlInclusiveNaming(is);
+                for (XmlData data : xml.getData()) {
+                    String tier = data.getTier();
+                    String term = data.getTerm();
+                    if (!ALLOWED_TIERS.contains(tier) && !inclusiveNamingConfig.excludeTerms().contains(term)) {
```

Comment:
Can `excludeTerms` be a regex or a wildcard matching expression ?

Reply:
Done, see on the first comment how to configure it in the pom